library/macos/FLETextInputModel.h

```diff
@@ -51,4 +51,28 @@ typedef NS_ENUM(NSUInteger, FLETextAffinity) { FLETextAffinityUpstream, FLETextA
  */
 @property(nonnull) NSDictionary *state;
 
+/**
+ * Id of the text input client.
+ */
+@property(nonatomic, readonly) NSNumber *clientID;
+
+/**
+ * Keyboard type of the client. See available options:
+ * https://docs.flutter.io/flutter/services/TextInputType-class.html
+ */
+@property(nonatomic, readonly) NSString *inputType;
```

Comment:
These need nullability annotations (this should be a warning in your local build). If you make them non-null, you'll need to make initWithClientID:configuration: either return nil if an option is missing, or have default values for them.